### PR TITLE
feat: unit tests and validations added for mark flag use case

### PR DESCRIPTION
**Pull request by danielglobant**
# Summary

## The issue

- Mark flag use case
- `flag_available` typo

### Principal

- Mark flag use case created using TDD

### Secondary

- `flag_available` typo replaced with  `flags_available` .
- `Models` group in the same folder.
- `WithBoardId` util created.

## Considerations of the solution

- Your feedback was implemented

## References

# Self review

- [ ] I have reviewed my PR changes and left comments/questions when necessary
- [ ] I have added a screenshot for the applied changes.
- [ ] I have followed the style guidelines of this project.
- [x] I have added and ran the tests locally.
- [ ] I have ran the test coverage reports locally.
- [x] I have made corresponding changes to the documentation (README and API).
- [ ] I have added the abstraction layers needed to separate responsibilities.

## Impact of these changes

This changes through `Component` impacts:

- `Component1`
- `Component2`
- `Component3`

## Evidences of expected changes

Add evidences here about each impact generated by your changes

# Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hot fix (fixes an issue in master branch)
- [ ] Docs (Documentation only changes)
- [ ] Style (Changes that do not affect the meaning of the code, formatting, missing semi-colons, etc)
- [ ] Refactor (Changes that neither fixes a bug or adds a feature)
- [ ] Performance (Changes that improve performance)
- [ ] Chore (Changes to the build process or auxiliary tools and libraries)

# Squash and merge

We are following [Angular commit messages guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines)

```
<type>(<scope>): <Jira-Issue subject>
<BLANK LINE>
<optional body>
<BLANK LINE>
<optional footer>
```

Types

```
feat: A new feature
fix: A bug fix
docs: Documentation only changes
style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
refactor: A code change that neither fixes a bug nor adds a feature
perf: A code change that improves performance
test: Adding missing or correcting existing tests
chore: Changes to the build process or auxiliary tools and libraries such as documentation generation
```

Make sure you squash and merge your feature/bugfix branches into develop/release/hotfix branches using the suggested format.

Some examples.

```
feat: DP-1234 Edit assessment information
Frontend changes for editing an assessment:
- improve assign modal to receive an existing assessment
- integrate the reasign end point.
- couple of warnings removed
BREAKING CHANGE: the assign modal now receives 3 required parameters
```

More examples [here](https://www.conventionalcommits.org/en/v1.0.0-beta.2/#summary)

### CODE REVIEW

File: minesweeper/use-cases/markFlag/markFlag.validations.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+export const makeValidations = (
+  props: IMarkFlagProps,
+  board: IMarkFlagResponse
+) => {
+  const [columnMarked, rowMarked] = props.position;
+  return {
+    OUTSIDE_BOARD: columnMarked > board.columns || rowMarked > board.rows,
+    ALREADY_A_FLAG: !!board.flags.find((flag) => flag === props),
```
Comment (Dylanflini) [inline minesweeper/use-cases/markFlag/markFlag.validations.ts:31]: this implementation could be more simple using a some method of Arrays.
For example: ALREADY_A_FLAG: board.flags.some((flag) => flag.position[0] === props.position[0] && flag.position[1] ===  props.position[1])

**Comment by Dylanflini** [comment]
Maybe the tests and implementation of IS_NATURAL_NUMBER could be mixed with the tests and implementation of OUTSITE_BOARD.
Because OUTSITE_BOARD references that it cannot be less than zero and greater than the columns or rows of board

File: minesweeper/infrastructure/data/index.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 import { IDataRepository } from '@minesweeper/domain/data.repository';
-import { createId } from '../dependencies/uuid';
+// import { createId } from '@minesweeper/infrastructure/dependencies/uuid';
 
 let db: BoardModel;
 
 export const dataRepository: IDataRepository = {
   saveBoard: async board => {
-    const id = createId();
+    const id = '111'; // createId(); error in jest test
```
Comment (danielglobant) [inline minesweeper/infrastructure/data/index.ts:9]: i need your help here @Dylanflini

File: minesweeper/use-cases/markFlag/markFlag.validations.ts (hunk 1 above)
Comment (danielglobant) [reply minesweeper/use-cases/markFlag/markFlag.validations.ts]: Done!!!

File: minesweeper/infrastructure/data/index.ts (hunk 2 above)
Comment (danielglobant) [reply minesweeper/infrastructure/data/index.ts:9]: I think is something of config

File: minesweeper/use-cases/markFlag/markFlag.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { BoardModel } from '@minesweeper/domain/Board.model';
+import { IBoardId, IPosition } from '@minesweeper/domain/commons.type';
+
+import { dataRepository } from '@minesweeper/infrastructure/data';
```
Comment (Dylanflini) [inline minesweeper/use-cases/markFlag/markFlag.ts:4]: The UseCases Layer can not depend of Infrastructure Layer, dataRepository should pass in props of markFlagUseCase function
```suggestion
import { IDataRepository } from '@minesweeper/domain/data.repository';
```

File: minesweeper/use-cases/markFlag/markFlag.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+import { makeValidations, MarkFlagUCError, GeneralError } from './markFlag.validations';
+
+export interface WithBoardId {
+  // export to commons
+  boardId: IBoardId;
+}
+
+export interface IMarkFlagProps extends IPosition, WithBoardId {}
```
Comment (Dylanflini) [inline minesweeper/use-cases/markFlag/markFlag.ts:13]:
```suggestion
export interface IMarkFlagProps extends IPosition, WithBoardId {
   dataRepository: IDataRepository;
}
```

File: minesweeper/use-cases/markFlag/markFlag.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+export type IMarkFlagResponse = BoardModel;
+export type IMarkFlagUseCase = (props: IMarkFlagProps) => Promise<IMarkFlagResponse>;
+
+/**
+ * Mark flag in board
+ */
+export const markFlagUseCase: IMarkFlagUseCase = async props => {
+  const { boardId, ...positionProps } = props;
```
Comment (Dylanflini) [inline minesweeper/use-cases/markFlag/markFlag.ts:22]:
```suggestion
  const { boardId, dataRepository, ...positionProps } = props;
```

**Pull request merged**